sw/device/lib/testing/meson.build

```diff
@@ -0,0 +1,78 @@
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+src_root_dir = meson.source_root()
+vendor_dir = src_root_dir / 'sw/vendor'
+gtest_src_dir = vendor_dir / 'google_googletest'
+gtest_build_dir = meson.current_build_dir() / 'google_googletest'
+gtest_lock_file = vendor_dir / 'google_googletest.lock.hjson'
+empty_file_for_dep = 'empty_file_for_googletest_dependency.cc'
+
+# Bash script used for building googletest. Also creates an empty file
+# which is used to establish the dependency between gtest and mock_mmio.
+build_gtest_cmd = '''
+echo "Building googletest..."
+GTEST_BUILD_DIR="@0@"
+GTEST_SRC_DIR="@1@"
+EMPTY_FILE_FOR_DEP="@2@"
+# Create a directory for building gtest
+mkdir -p $GTEST_BUILD_DIR
+# Build googletest
+cmake -B $GTEST_BUILD_DIR -S $GTEST_SRC_DIR > /dev/null
+make -C $GTEST_BUILD_DIR > /dev/null
+# Create an empty file for dependencies
+touch "@OUTDIR@/$EMPTY_FILE_FOR_DEP"
+echo "Done!"
+'''.format(gtest_build_dir, gtest_src_dir, empty_file_for_dep)
+
+build_gtest = custom_target(
+  'googletest',
+  output: empty_file_for_dep,
+  depend_files: gtest_lock_file,
+  command: ['bash', '-c', build_gtest_cmd],
+)
+
+gtest_inc = include_directories(
+  '../../../vendor/google_googletest/googletest/include',
+  '../../../vendor/google_googletest/googlemock/include',
+)
+
+gtest = declare_dependency(
+  sources: [build_gtest],
+  include_directories: gtest_inc,
+  link_args: [
+    '-L' + gtest_build_dir / 'lib',
+    '-lgmock',
+    '-lgmock_main',
+    '-lgtest',
+    '-lpthread'
```

Comment:
Doesn't meson have some arg for magically causing pthread to get linked in (like "threads" or something?). I suspect it's flaky but you should give it a try just in case.

Reply:
Done: `dependency('threads')`.